packages/flutter_tools/test/general.shard/base/file_system_test.dart

```diff
@@ -57,6 +58,44 @@ void main() {
     });
   });
 
+  testWithoutContext('deleteIfExists does not delete if file does not exist', () {
```

Comment:
If the extension method applies to any implementation of `File`, then we need test coverage of how it applies to them.